### PR TITLE
Add missing except+ to Cython bindings

### PR DESCRIPTION
**Pull request by shwina**

### CODE REVIEW

File: python/cudf/cudf/_libxx/copying.pyx [hunk 1]
```diff
@@ -21,7 +21,7 @@ def gather(Table source_table, Column gather_map):
     cdef column_view gather_map_view = gather_map.view()
 
     with nogil:
-        c_result = (
+        c_result = move(
```
Comment (trxcllnt) [inline python/cudf/cudf/_libxx/copying.pyx]: Is this `move()` call required in addition to the `move()` for the `Table.from_unique_ptr()` call below?

Comment (shwina) [reply python/cudf/cudf/_libxx/copying.pyx]: I found that I needed this the moment I put in `except+` -- no idea why that happened.

Comment (cwharris) [reply python/cudf/cudf/_libxx/copying.pyx]: I'm just guessing here, but it sounds like adding `except+` introduces a new variables transparently, but assigns the new variable using some amalgamation of the provided expression. The return value of the function is usually move-assigned by default, but the introduction of an intermediate variable would require the next assignment to be explicit in it's "move"-ness.

```python
the_result
try:
 the_result = gather(...) # move is implicit
c_result = the_result # move must be explicit
```

Comment (kkraus14) [reply python/cudf/cudf/_libxx/copying.pyx]: We could check the generated cpp for confirmation if we want 😄

**Pull request merged**